### ISSUE TITLE
Build fails with example project and geolocator 3.0.0

### ISSUE THREAD

**Issue by Pitazzo** [BaseflowIT/flutter-geolocator#220]
## 🐛 Bug Report

Just after adding the geolocator dependecy to the default flutter project, it's impossible to build the app. If I try to launch it in debug mode, it just doesn't do anything and the app doesn't open in my device. When trying to build it using `flutter build apk`, I get the following stacktrace: 

```
Initializing gradle...                                              1,1s
Resolving dependencies...                                           2,0s
Note: Some input files use unchecked or unsafe operations.              
Note: Recompile with -Xlint:unchecked for details.                      
D8: Program type already present: android.support.v4.os.ResultReceiver$MyResultReceiver
                                                                        
FAILURE: Build failed with an exception.                                
                                                                        
* What went wrong:                                                      
Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForRelease'.
> com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/0.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/1.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/2.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/3.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/4.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/5.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/6.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/7.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/8.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/9.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/10.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/11.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/12.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/13.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/14.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/15.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/16.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/17.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/18.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/19.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/20.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/21.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/22.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/23.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/24.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/25.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/26.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/27.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/28.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/29.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/30.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/31.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/32.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/33.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/34.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/35.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/36.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/37.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/38.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/39.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/40.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/41.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/42.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/43.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/44.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/45.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/46.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/47.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/48.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/49.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/50.jar, /home/pitazzo/Proyectos/location_test/build/app/intermediates/transforms/dexBuilder/release/51.jar
  Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
  Program type already present: android.support.v4.os.ResultReceiver$MyResultReceiver
                                                                        
* Try:                                                                  
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
                                                                        
* Get more help at https://help.gradle.org                              
                                                                        
BUILD FAILED in 7s                                                      
Running Gradle task 'assembleRelease'...                                
Running Gradle task 'assembleRelease'... Done                       8,3s
*******************************************************************************************
The Gradle failure may have been because of AndroidX incompatibilities in this Flutter app.
See https://goo.gl/CP92wY for more information on the problem and how to fix it.
*******************************************************************************************
Gradle task assembleRelease failed with exit code 1

```

### Expected behavior
Regular build of my application
### Reproduction steps
Just created a plain flutter example project and added the geolocator dependecy to the `pubspec.yaml` file

### Configuration
Default configuration
**Version:** 3.0.0

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android

**Comment by jkyon** [comment]
I have the same issue in my app. Is there a workaround to fix this?

thank you in advance

**Comment by jkyon** [comment]
I found a workaround, and it works for me. Check this post on StackOverflow
https://stackoverflow.com/questions/50387207/error-while-merging-dex-program-type-already-present-android-support-v4-os-resu

**Comment by Pitazzo** [comment]
> I have the same issue in my app. Is there a workaround to fix this?
> 
> thank you in advance

As far as I know, I can't find a way to get it working

> I found a workaround, and it works for me. Check this post on StackOverflow
> https://stackoverflow.com/questions/50387207/error-while-merging-dex-program-type-already-present-android-support-v4-os-resu

 This doesn't work for me. What did you exactly do? I have added
```
android.useAndroidX=true
android.enableJetifier=true
```
to my `gradle.properties ` and those dependencies to my `build.gradle` file. However, I'm still not able to build the example app and getting this stacktrace:

```
Initializing gradle...                                              0,9s
Resolving dependencies...                                           1,7s
/home/pitazzo/SDKs/flutter/.pub-cache/hosted/pub.dartlang.org/geolocator-2.1.1/android/src/main/java/com/baseflow/flutter/plugin/geolocator/tasks/LocationUpdatesUsingLocationServicesTask.java:5: error: package android.support.annotation does not exist
import android.support.annotation.NonNull;                              
                                 ^                                      
/home/pitazzo/SDKs/flutter/.pub-cache/hosted/pub.dartlang.org/geolocator-2.1.1/android/src/main/java/com/baseflow/flutter/plugin/geolocator/tasks/LastKnownLocationUsingLocationServicesTask.java:4: error: package android.support.annotation does not exist
import android.support.annotation.NonNull;                              
                                 ^                                      
/home/pitazzo/SDKs/flutter/.pub-cache/hosted/pub.dartlang.org/geolocator-2.1.1/android/src/main/java/com/baseflow/flutter/plugin/geolocator/tasks/LocationUpdatesUsingLocationServicesTask.java:62: error: cannot find symbol
                    public void onComplete(@NonNull Task<Void> task) {  
                                            ^                           
  symbol: class NonNull                                                 
/home/pitazzo/SDKs/flutter/.pub-cache/hosted/pub.dartlang.org/geolocator-2.1.1/android/src/main/java/com/baseflow/flutter/plugin/geolocator/tasks/LastKnownLocationUsingLocationServicesTask.java:40: error: cannot find symbol
                    public void onFailure(@NonNull Exception e) {       
                                           ^                            
  symbol: class NonNull                                                 
Note: Some input files use unchecked or unsafe operations.              
Note: Recompile with -Xlint:unchecked for details.                      
4 errors                                                                
                                                                        
FAILURE: Build failed with an exception.                                
                                                                        
* What went wrong:                                                      
Execution failed for task ':geolocator:compileReleaseJavaWithJavac'.    
> Compilation failed; see the compiler error output for details.        
                                                                        
* Try:                                                                  
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
                                                                        
* Get more help at https://help.gradle.org                              
                                                                        
BUILD FAILED in 1s                                                      
Running Gradle task 'assembleRelease'...                                
Running Gradle task 'assembleRelease'... Done                       1,8s
*******************************************************************************************
The Gradle failure may have been because of AndroidX incompatibilities in this Flutter app.
See https://goo.gl/CP92wY for more information on the problem and how to fix it.
*******************************************************************************************
Gradle task assembleRelease failed with exit code 1
```

**Comment by jkyon** [comment]
updates geolocator: ^3.0.0

**Comment by fyzuladzwan** [comment]
Hi. I couldnt get this work either. I think the google_api_availability is holding everything back. The geolocator is depending on google_api_availability, which is still stuck in the old pre-AndroidX migration. Hope this helps to whoever can solve it.

**Comment by tayormi** [comment]
I am facing this issue as well.

**Comment by kyleodev** [comment]
The plugin needs to be updated. Facing the same issue.

**Comment by mleonhard** [comment]
Android builds were failing on my machine with:
```
Program type already present: android.support.v4.os.ResultReceiver$MyResultReceiver
```

Adding these lines to android/gradle.properties solved it:
```
android.useAndroidX=true
android.enableJetifier=true
```

**Comment by dokinkon** [comment]
please migrate google_api_availability to 2.0.0, thanks.

**Comment by mvanbeusekom** [comment]
We just released version 3.0.1 of the Geolocator plugin which takes a dependency on the `google_api_availability` plugin version 2.0.0. 

this should solve this problem together with making sure you Flutter project is migrated to support AndroidX (which it doesn't out of the box). For instructions [see](https://github.com/BaseflowIT/flutter-geolocator/issues/196#issuecomment-465460801)

**Comment by paapu88** [comment]
I was using 3.0.1 and got the same error as mleonhard. His solution worked for me.
